File: src/main/java/com/squarepolka/digiscope/plot/PlotPointIterator.java
package com.squarepolka.digiscope.plot;

import com.squarepolka.digiscope.plot.plotpoint.PlotPoint;

import java.util.Iterator;

public class PlotPointIterator implements Iterator<PlotPoint> {

    private PlotPoint nextPlotPoint;

    public PlotPointIterator(PlotPoint firstPlotPoint) {
        this.nextPlotPoint = firstPlotPoint;
    }

    public boolean hasNext() {
        return null != nextPlotPoint;
    }

    public PlotPoint next() {
        PlotPoint plotPointToReturn = nextPlotPoint;
        nextPlotPoint = nextPlotPoint.getNextPlotPoint();
        return plotPointToReturn;
    }

}

File: src/main/kotlin/recording/PlotPointIterator.kt
package recording

import plotpoint.PlotPoint

class PlotPointIterator (private var nextPlotPoint: PlotPoint?) : Iterator<PlotPoint> {

    override operator
    fun hasNext(): Boolean {
        return null != nextPlotPoint
    }

    override operator
    fun next(): PlotPoint {
        val plotPointToReturn = nextPlotPoint ?: PlotPoint()
        nextPlotPoint = nextPlotPoint?.nextPlotPoint
        return plotPointToReturn
    }

}

